site/src/pages/WorkspacePage/WorkspaceActions/constants.ts

```diff
@@ -1,59 +1,76 @@
-import { Workspace, WorkspaceStatus } from "api/typesGenerated";
-import { ReactNode } from "react";
+import { type Workspace, type WorkspaceStatus } from "api/typesGenerated";
 import { workspaceUpdatePolicy } from "utils/workspace";
 
-// the button types we have
-export enum ButtonTypesEnum {
-  start = "start",
-  starting = "starting",
-  stop = "stop",
-  stopping = "stopping",
-  restart = "restart",
-  restarting = "restarting",
-  deleting = "deleting",
-  update = "update",
-  updating = "updating",
-  activate = "activate",
-  activating = "activating",
-  // disabled buttons
-  canceling = "canceling",
-  deleted = "deleted",
-  pending = "pending",
-}
+/**
+ * An iterable of all button types supported by the workspace actions UI
+ */
+const buttonTypes = [
+  "start",
+  "starting",
+  "stop",
+  "stopping",
+  "restart",
+  "restarting",
+  "deleting",
+  "update",
+  "updating",
+  "activate",
+  "activating",
 
-export type ButtonMapping = {
-  [key in ButtonTypesEnum]: ReactNode;
-};
+  // There's no need for a retrying state because retrying starts a transition
+  // into one of the starting, stopping, or deleting states (based on the
+  // WorkspaceTransition type)
+  "retry",
+  "retryDebug",
+
+  // These are buttons that should be used with disabled UI elements
+  "canceling",
+  "deleted",
+  "pending",
+] as const;
 
-interface WorkspaceAbilities {
-  actions: ButtonTypesEnum[];
+/**
+ * A button type supported by the workspace actions UI
+ */
+export type ButtonType = (typeof buttonTypes)[number];
```

Comment:
I'm not familiar with this syntax...what is going on here?

Reply:
Appreciate the explanation. It would be nice to use this syntax with `TypesGen.WorkspaceTransitions` but idk how to make that happen on the Go side